protocol.md

```diff
@@ -726,6 +727,120 @@ Upload-Length: 11
 Upload-Concat: final;/files/a /files/b
 ```
 
+### Client Tag
+
+With this extension, Clients can provide a Tag for an upload - with which they
+can later retrieve the URL for the created resource.
+
+This allows Clients to resume an upload initiated through the [Creation With Upload](#creation-with-upload)
+or [Creation](#creation) extensions for which they did not receive the response with
+the newly created resource's URL (for example due to a broken connection).
+
+If the Server supports this extension, it MUST add `client-tag` to the `Tus-Extension`
+header.
+
+To initiate an upload with Client Tag, Clients MUST include an `Upload-Tag`
+header with the initial `POST` request (as specified in the [Creation](#creation) and
+[Creation With Upload](#creation-with-upload) extensions) made to the Upload URL.
+
+Clients can then send a `HEAD` request with the same `Upload-Tag` header to the
+Upload URL, which - upon success - MUST respond with the `200 OK` or `204 No Content`
+status, and the resource's URL in the response's `Location` header.
+
+A successful response MUST also contain the `Tus-Version` header. An `Upload-Offset` header
+that indicates how many bytes have already been received by the server MAY also be returned.
+
+Clients can then use the returned `Location` to resume the upload.
+
+If an `Upload-Tag` is unknown or no longer known by the Server, it MUST
+respond to `HEAD` requests with the `404 Not Found` or `410 Gone` status.
+
+If an `Upload-Tag` is already in use for a different, ongoing upload, the Server
+MUST respond to the `POST` request initiating the upload with a `409 Conflict` status.
+
+#### Security Considerations
+
+To protect against attackers trying to guess an `Upload-Tag` and subsequently
+use it to inject malicious content into an upload, Servers MUST take measures to
+ensure that only the party that created a resource with an `Upload-Tag` can use it
+for subseqeuent `PATCH` and `HEAD` requests.
+
+If uploading to a Server is only possible for authenticated users, the Server can
+satisfy this requirement by leveraging the available authentication information to
+bind the `Upload-Tag` to a particular user, so that only that user can use it.
+
+Servers MUST return a `403 Forbidden` status for unauthenticated requests that
+contain an `Upload-Tag` header.
```

Comment:
I'm not really sure how to phrase it but I think that we need to open this sentence up a bit to make it possible to add the challenge extension later. A client using the challenge extension might be unauthenticated but should still be able to access the file, given that they provide the the correct Upload-Challenge. 

> Servers MUST return a `403 Forbidden` status for unauthenticated requests that
> contain an `Upload-Tag` header unless the server can guarantee that the client making the request is the one that created the resource.

Reply:
I agree that this sentence is too restrictive. We should recommend server to check authentication but we should not require them to do so. For example, tusd does not handle authentication at all because it delegates that to the application itself. If we put this sentence into the specification, tusd would not be able to implement the specification.

Reply:
I tried to find a less specific wording:

> Servers MUST return a `403 Forbidden` status response for requests containing an
> `Upload-Tag` header if the request is not attributable to the original creator
> of the referenced upload resource.

What do you think?

Reply:
Update: the next commit will [remove/replace](https://github.com/tus/tus-resumable-upload-protocol/pull/163#discussion_r493076235) that paragraph.